modules/openy_features/openy_block/modules/openy_block_branch_amenities/src/Plugin/Block/AmenitiesWithIcons.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Drupal\openy_block_branch_amenities\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Cache\Cache;
+use Drupal\node\NodeInterface;
+
+/**
+ * Provides a 'Node Amenities' block.
+ *
+ * @Block(
+ *   id = "branch_amenities_icons",
+ *   admin_label = @Translation("Branch Amenities with icons Block"),
+ *   category = @Translation("Paragraph Blocks")
+ * )
+ */
+class AmenitiesWithIcons extends BlockBase {
+
+  /**
+   * @todo add cache
+   * {@inheritdoc}
+   */
+  public function build() {
+    $render_array = [];
+
+    $node = \Drupal::routeMatch()->getParameter('node');
```

Comment:
This should be implemented using Dependency Injection

Reply:
Example: https://www.drupal.org/docs/8/api/services-and-dependency-injection/dependency-injection-in-plugin-block